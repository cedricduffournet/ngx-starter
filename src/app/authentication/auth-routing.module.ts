import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent
} from '@app/authentication/containers';
import { authConfig } from '@app/authentication/auth.config';

const AUTHENTICATION_ROUTES: Routes = [
  { path: 'login', component: LoginViewComponent },
  { path: 'resetting', component: SendResetPasswordViewComponent },
  {
    path: `${authConfig.path.passwordResetConfirm}/:token`,
    component: ResetPasswordViewComponent
  },
  { path: 'register', component: RegisterViewComponent },
  {
    path: `${authConfig.path.registerConfirm}/:token`,
    component: RegisterConfirmViewComponent
  }
];

@NgModule({
  imports: [RouterModule.forChild(AUTHENTICATION_ROUTES)],
  exports: [RouterModule]
})
export class AuthRoutingModule {}
