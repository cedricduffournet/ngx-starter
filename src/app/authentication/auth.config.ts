export const authConfig = {
  path: {
    registerConfirm: 'register/confirm',
    passwordResetConfirm: 'password/reset'
  },
  storageKey: 'authentication'
};
