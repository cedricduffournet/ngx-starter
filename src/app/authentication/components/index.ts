export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
