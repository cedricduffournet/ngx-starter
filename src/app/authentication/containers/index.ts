export * from './login-view/login-view.component';
export * from './register-confirm-view/register-confirm-view.component';
export * from './register-view/register-view.component';
export * from './reset-password-view/reset-password-view.component';
export * from './send-reset-password-view/send-reset-password-view.component';
export * from './logout-modal/logout-modal.component';
