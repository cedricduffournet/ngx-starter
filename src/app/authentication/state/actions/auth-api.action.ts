import { props, createAction } from '@ngrx/store';

import { User } from '@app/user/models/User';
import { AuthToken } from '@app/authentication/models/auth';

export const loginSuccess = createAction(
  '[Auth/API] Login success',
  props<{ authToken: AuthToken }>()
);

export const loginFailure = createAction(
  '[Auth/API] Login failure',
  props<{ error: any }>()
);

export const registerSuccess = createAction(
  '[Auth/API] Register success',
  props<{ userRegistered: User }>()
);

export const registerFailure = createAction(
  '[Auth/API] Register failure',
  props<{ error: any }>()
);

export const loadLoggedUserSuccess = createAction(
  '[Auth/API] Load logged user success',
  props<{ user: User }>()
);

export const loadLoggedUserFailure = createAction(
  '[Auth/API] Load logged user failure',
  props<{ error: any }>()
);

export const loadTokenSuccess = createAction(
  '[Auth/API] load token success',
  props<{ authToken: AuthToken }>()
);

export const loadTokenFailure = createAction(
  '[Auth/API] load token failure',
  props<{ error: string }>()
);

export const logoutSuccess = createAction('[Auth] log out success');
