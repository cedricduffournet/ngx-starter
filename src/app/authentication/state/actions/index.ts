import * as AuthActions from './auth.actions';
import * as AuthApiActions from './auth-api.action';
import * as LoginViewActions from './login-view.actions';
import * as RegisterViewActions from './register-view.actions';

export { AuthActions, AuthApiActions, LoginViewActions, RegisterViewActions };
