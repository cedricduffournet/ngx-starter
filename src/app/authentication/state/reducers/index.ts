import {
  createSelector,
  createFeatureSelector,
  Action,
  combineReducers
} from '@ngrx/store';

import { User } from '@app/user/models/User';
import * as fromRoot from '@app/core/state/reducers';
import * as fromAuth from '@app/authentication/state/reducers/auth.reducer';
import * as fromLoginView from '@app/authentication/state/reducers/login-view.reducer';
import * as fromRegisterView from '@app/authentication/state/reducers/register-view.reducer';

export interface AuthState {
  userConnected: fromAuth.State;
  loginView: fromLoginView.State;
  registerView: fromRegisterView.State;
}

export interface State extends fromRoot.State {
  auth: AuthState;
}

export function reducers(state: AuthState | undefined, action: Action) {
  return combineReducers({
    userConnected: fromAuth.reducer,
    loginView: fromLoginView.reducer,
    registerView: fromRegisterView.reducer
  })(state, action);
}

export const selectAuthState = createFeatureSelector<State, AuthState>('auth');

export const getUserConnectedState = createSelector(
  selectAuthState,
  auth => auth.userConnected
);
export const getLoggedUser = createSelector(
  getUserConnectedState,
  auth => auth.user
);
export const getIsLogged = createSelector(
  getLoggedUser,
  user => !!user
);

export const getAuthorized = createSelector(
  getLoggedUser,
  (user: User | null, props: any) => {
    if (!user) {
      return false;
    }
    return props.roles.some((element: string) => {
      return user.roles.some((element2: string) => {
        return element2 === element;
      });
    });
  }
);

export const selectLoginViewState = createSelector(
  selectAuthState,
  (state: AuthState) => state.loginView
);
export const getLoginViewError = createSelector(
  selectLoginViewState,
  fromLoginView.getError
);
export const getLoginViewProcessing = createSelector(
  selectLoginViewState,
  fromLoginView.getProcessing
);

export const selectRegisterViewState = createSelector(
  selectAuthState,
  (state: AuthState) => state.registerView
);
export const getRegisterViewProcessing = createSelector(
  selectRegisterViewState,
  fromRegisterView.getProcessing
);
export const getRegisterViewStatus = createSelector(
  selectRegisterViewState,
  fromRegisterView.getStatus
);
export const getRegisterViewErrorUserExist = createSelector(
  selectRegisterViewState,
  fromRegisterView.getErrorUserExist
);
export const getRegisterViewUserRegistered = createSelector(
  selectRegisterViewState,
  fromRegisterView.getUserRegistered
);
