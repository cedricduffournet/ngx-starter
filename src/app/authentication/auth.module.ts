import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityModule } from '@app/civility/civility.module';
import { LangSwitcherModule } from '@app/shared/lang-switcher/lang-switcher.module';
import { LogoModule } from '@app/shared/logo/logo.module';
import { AuthRoutingModule } from '@app/authentication/auth-routing.module';
import { AuthEffects } from '@app/authentication/state/effects';
import { reducers } from '@app/authentication/state/reducers';
import { AuthFacade } from '@app/authentication/state/auth.facade';

import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent,
  LogoutModalComponent
} from '@app/authentication/containers';

import {
  AuthLayoutComponent,
  LoginFormComponent,
  ResetPasswordFormComponent,
  SendResetPasswordFormComponent,
  RegisterFormComponent
} from '@app/authentication/components';

@NgModule({
  imports: [
    AuthRoutingModule,
    SharedModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
    LangSwitcherModule,
    LogoModule,
    CivilityModule
  ],
  declarations: [
    AuthLayoutComponent,
    LoginViewComponent,
    SendResetPasswordViewComponent,
    ResetPasswordViewComponent,
    RegisterViewComponent,
    RegisterConfirmViewComponent,
    LoginFormComponent,
    ResetPasswordFormComponent,
    SendResetPasswordFormComponent,
    LogoutModalComponent,
    RegisterFormComponent
  ],
  entryComponents: [LogoutModalComponent],
  providers: [AuthFacade]
})
export class AuthModule {}
