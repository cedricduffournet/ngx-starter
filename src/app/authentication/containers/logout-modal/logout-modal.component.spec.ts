import { TestBed, ComponentFixture } from '@angular/core/testing';

import { Store } from '@ngrx/store';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { BsModalRef } from 'ngx-bootstrap/modal';
import { TranslateModule } from '@ngx-translate/core';

import { ValidationActionModule } from '@app/shared/validation-action';
import { ModalWrapperModule } from '@app/shared/modal';
import { LogoutModalComponent } from '@app/authentication/containers';
import * as fromAuth from '@app/authentication/state/reducers';
import { AuthActions } from '@app/authentication/state/actions';

describe('LogoutConfirm', () => {
  let fixture: ComponentFixture<LogoutModalComponent>;
  let component: LogoutModalComponent;
  let store: MockStore<fromAuth.State>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LogoutModalComponent],
      imports: [
        TranslateModule.forRoot(),
        ModalWrapperModule,
        ValidationActionModule
      ],
      providers: [provideMockStore(), BsModalRef]
    });

    fixture = TestBed.createComponent(LogoutModalComponent);
    component = fixture.componentInstance;
    store = TestBed.get(Store);
    spyOn(store, 'dispatch');
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should call closeModal on cancel', () => {
    spyOn(component, 'closeModal');
    component.onCancel();
    expect(component.closeModal).toHaveBeenCalled();
  });

  it('should dispatch AuthActions.logout on confirm', () => {
    const action = AuthActions.logout();
    component.onConfirm();
    expect(store.dispatch).toHaveBeenCalledWith(action);
  });

  it('should call closeModal on confirm', () => {
    spyOn(component, 'closeModal');
    component.onConfirm();
    expect(component.closeModal).toHaveBeenCalled();
  });

  it('should hide modal when closeModal is called', () => {
    spyOn(component.bsModalRef, 'hide');
    component.closeModal();
    expect(component.bsModalRef.hide).toHaveBeenCalled();
  });
});
