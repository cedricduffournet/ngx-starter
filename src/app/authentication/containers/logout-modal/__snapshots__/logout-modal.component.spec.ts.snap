// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutConfirm should be created 1`] = `
<app-logout-modal
  bsModalRef={[Function BsModalRef]}
  store={[Function MockStore]}
>
  <app-modal-wrapper
    ng-reflect-title="CONFIRM"
  >
    <div
      class="modal-content bg-light"
    >
      
      <div
        class="modal-header"
      >
        <h4
          class="modal-title"
        >
          CONFIRM
        </h4>
      </div>
      <div
        class="modal-body p-0"
      >
        <section
          class="bg-white"
        >
          <div
            class="p-3"
            ng-reflect-translate=""
            translate=""
          >
             CONFIRM_LOGOUT 
          </div>
        </section>
        <app-wrapper-validation-action>
          <div
            class="d-flex justify-content-end align-items-center p-3"
          >
            <app-validation-button
              ng-reflect-label-cancel="NO"
              ng-reflect-label-confirm="YES"
            >
              <div
                class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
                ng-reflect-klass="button-validation-action text-"
                ng-reflect-ng-class=""
              >
                
                <button
                  class="btn btn-cancel btn-lg btn-light"
                  ng-reflect-klass="btn btn-cancel btn-lg"
                  ng-reflect-ng-class="btn-light"
                  type="button"
                >
                   NO 
                </button>
                
                <button
                  class="btn btn-confirm btn-lg btn-primary"
                  ng-reflect-klass="btn btn-confirm btn-lg"
                  ng-reflect-ng-class="btn-primary"
                  type="submit"
                >
                  
                  
                  
                  <span>
                    YES
                  </span>
                </button>
              </div>
            </app-validation-button>
          </div>
        </app-wrapper-validation-action>
      </div>
    </div>
  </app-modal-wrapper>
</app-logout-modal>
`;
